### PR TITLE
[GraphQL] Only exposes the mutations if any

### PR DESCRIPTION
**Pull request by sroze**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | ø
| License       | MIT
| Doc PR        | ø

Otherwise, GraphiQL gives this lovely message:
> Mutation fields must be an object with field names as keys or a function which returns [...]

### CODE REVIEW

File: src/GraphQl/Type/SchemaBuilder.php [hunk 1]
```diff
@@ -92,16 +92,21 @@ public function getSchema(): Schema
...
-        return new Schema([
+        $schema = [
             'query' => new ObjectType([
                 'name' => 'Query',
                 'fields' => $queryFields,
             ]),
-            'mutation' => new ObjectType([
+        ];
+
+        if (\count($mutationFields) > 0) {
```
Comment (dunglas) [inline src/GraphQl/Type/SchemaBuilder.php]: `if ($mutationFields)`?

**Comment by alanpoulain** [comment]
Maybe you could add a unit test or a Behat one (by using introspection)?

**Comment by sroze** [comment]
Rebased and changed check with a simple `if`.

@alanpoulain based on the complexity of the `SchemaBuilderTest`, I don't think it's worth it. Also, I'll follow up with a fair amount of pull-requests regarding the messenger integration in the next few weeks that should cover this use-case :)

**Comment by dunglas** [comment]
Thanks @sroze

**Pull request merged**